python/ray/data/datasource/file_based_datasource.py

```diff
@@ -155,6 +155,15 @@ def __init__(
         file_extensions: Optional[List[str]] = None,
     ):
         _check_pyarrow_version()
+
+        self._supports_distributed_reads = not _is_local_scheme(paths)
+        if not self._supports_distributed_reads and ray.util.client.ray.is_connected():
```

Comment:
if it's straightforward, maybe include another unit test to check the case with local path + ray client connected to assert the error is raised?